include/ranges.h

```diff
@@ -64,6 +66,17 @@ struct subrange {
 	subrange(chunk<Dims> other) : offset(other.offset), range(other.range) {}
 
 	bool operator==(const subrange& rhs) { return offset == rhs.offset && range == rhs.range; }
+
+	std::string to_string() const {
+		if(dims == 1)
+			return fmt::format("subrange({}) at offset({})", range[0], offset[0]);
+		else if(dims == 2)
+			return fmt::format("subrange({},{}) at offset({},{})", range[0], range[1], offset[0], offset[1]);
+		else
+			return fmt::format("subrange({},{},{}) at offset({},{},{})", range[0], range[1], range[2], offset[0], offset[1], offset[2]);
+	}
```

Comment:
I recently merged a commit by @PeterTh (3ff6530) that introduces generic printing capabilities for `celerity::subrange` and `celerity::chunk`. You should be able to use those for your purposes.

Reply:
Not trying to print, trying to log. Unfortunately, the commit does not cater for that.